rearranging_cars.py

```diff
@@ -0,0 +1,53 @@
+from collections import OrderedDict
+
+def rearranging_cars(old_parking, new_parking):
+    """
+    Input: two lists of the parking slots such as in every slot exist the number of the car from 0 to N-1.
+    Where 0 is the empty slot.
+    old_parking : the old positions
+    new_parking : the new positions
+    Output: list of tuples of needed moves such as every tuple is (from , to).
+    """
+    steps = []
+    # the keys are the number of car and the values are the positions of the cars
+    map_slot_cars = OrderedDict()
+    
+    for slot_index in range(len(old_parking)):
+        car_num = old_parking[slot_index]
+        map_slot_cars[car_num] = slot_index
+
+    for new_slot_position in range(len(new_parking)):
+        empty_park_position = map_slot_cars[0]
+        new_park = new_parking[new_slot_position]
+        old_park_position = map_slot_cars[new_park]        
+        if new_slot_position != old_park_position and new_parking[new_slot_position] != 0:
+            #checking if there is a need for an extra move of the wanted park in case it is full
+            if new_slot_position != empty_park_position:
+                old_parking[empty_park_position] = old_parking[new_slot_position]
+                map_slot_cars[old_parking[new_slot_position]] = empty_park_position
+                steps.append((new_slot_position, empty_park_position))
+                
+            old_parking[new_slot_position] = new_park
+            old_parking[old_park_position] = 0
+            map_slot_cars[new_park] = new_slot_position
+            map_slot_cars[0] = old_park_position
+            steps.append((old_park_position, new_slot_position))
+        
```

Comment:
```
    final_arrangement = []
    for i in range(len(old_parking)):
        final_arrangement.append(map_slot_cars[i])
    return final_arrangement
```

If I add this code here, should it return the right arrangement?
If yes, it fails on this input:

init: [6, 10, 2, 3, 8, 4, 5, 7, 1, 0, 9]
suggested output: [2, 1, 10, 8, 7, 3, 6, 9, 0, 4, 5]
actual output: [8, 1, 0, 5, 9, 10, 6, 4, 3, 7, 2]